### PR TITLE
Reword kdump proposal warning

### PR DESCRIPTION
**Pull request by dgdavid**
## Problem

The kdump proposal warning is a bit confusing because it does not point out that kdump is the reason of the space requirement

>  Warning! There might not be enough free space. 104 GiB required, but only 29.86 GiB are available.

- https://bugzilla.suse.com/show_bug.cgi?id=1095207
- https://trello.com/c/Bgk82UaA/

## Solution

Reword the warning a bit to make it more clear.

> Warning! There might not be enough free space **to have kdump enabled**. 104 GiB required **for saving a kernel dump**, but only 29.86 GiB are available.

## Tests

Unit test for the Kdump client was broken. To fix it, the client has been moved to `lib/` and slightly adapted to make it more testable, allowing us to mockup not needed methods coming from `Yast::KdumpComplexIncludes` routines that were triggering not relevant dialogs during testing.

In the middle of the process, other unit test were relocated. If it hurts, those commits can be reverted.

### CODE REVIEW

**Review by imobachgs — APPROVED** [review]

**Comment by dgdavid** [comment]
@imobachgs, please review it again. See `Tests` section in the PR description.

File: test/kdump_client_test.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
+require_relative "./test_helper"
+
+# Let's "dirty-mock" Yast::KdumpComplexInclude to avoid not desired dialogs during test
```
Comment (imobachgs) [inline test/kdump_client_test.rb:3]: Please, this hack deserves a FIXME note :-)

Comment (dgdavid) [reply test/kdump_client_test.rb:3]: Sure, although I don't know what say there :P

**Review by imobachgs — APPROVED** [review]
LGTM

File: .rubocop.yml [hunk 2]
```diff
@@ -22,7 +22,6 @@ Metrics/BlockNesting:
   Include:
     - 'src/lib/**/*.rb' # be more strict for new code in lib
 
-
```
[the remark this answers is not in the document]
Comment (dgdavid) [reply .rubocop.yml:25]: Yes, you're right. Anyway, let's keep it there.

**Pull request merged**